include/roboteam_ai/world/FieldMessage.h

```diff
@@ -14,124 +10,234 @@
+     * goal area. */
+    TOP_LEFT_PENALTY_STRETCH,
+    /* The line closest (of all our goal area lines) to the bottom line and parallel to the bottom line. This line marks
+     * our goal area. */
+    BOTTOM_LEFT_PENALTY_STRETCH,
+    /* The line closest (of all the opponents goal area lines) to the top line and parallel to the top line. This line
+     * marks the opponents goal area. */
+    TOP_RIGHT_PENALTY_STRETCH,
+    /* The line closest (of all the opponents goal area lines) to the bottom line and parallel to the bottom line. This
+     * line marks the opponents goal area. */
+    BOTTOM_RIGHT_PENALTY_STRETCH
+};
+
+enum FieldVectorName {
+    OUR_GOAL_CENTER, // The middle point of our goal (this point is on the left line).
+    THEIR_GOAL_CENTER, // The middle point of the opponents goal (this point is on the right line).
+    LEFT_PENALTY_POINT, // The penalty point from which penalties are made towards our goal.
+    RIGHT_PENALTY_POINT, // The penalty point from which penalties are made towards the opponents goal.
+    OUR_BOTTOM_GOAL_SIDE, // The bottom most point of our goal (this point is on the left line).
+    OUR_TOP_GOAL_SIDE, // The top most point of our goal (this point is on the left line).
+    THEIR_BOTTOM_GOAL_SIDE, // The bottom most point of the opponents goal (this point is on the right line).
+    THEIR_TOP_GOAL_SIDE, // The top most point of the opponents goal (this point is on the right line).
 };
 
+enum FieldArcName {
+    CENTER_CIRCLE // The circle in the middle from which the ball will be kicked off
+};
 
+/**
+ * Stores all singular constant data (data that does not change through the match) about the field, which includes: <br>
+ * - Length, widths, heights of the field, goals and boundary. <br>
+ * - The location, direction and sizes of all lines on the field. <br>
+ * - The location and sizes of all arcs on the field.
+ * @author Created by: Lukas Bos <br>
+ *         Documented by: Haico Dorenbos
+ * @since 2019-08-30
+ */
 class FieldMessage {
-  FRIEND_TEST(FieldTest, line_intersects_with_defence_area);
-  FRIEND_TEST(FieldTest, it_gets_points_in_defence_area);
-  FRIEND_TEST(FieldTest, it_returns_proper_goal_centers);
-  FRIEND_TEST(FieldTest, it_detects_points_in_field_properly);
-  FRIEND_TEST(FieldTest, it_calculates_obstacles);
-  FRIEND_TEST(FieldTest, penalty_points);
-  FRIEND_TEST(FieldTest, goal_angle);
+    FRIEND_TEST(FieldTest, line_intersects_with_defence_area);
+    FRIEND_TEST(FieldTest, it_gets_points_in_defence_area);
+    FRIEND_TEST(FieldTest, it_returns_proper_goal_centers);
+    FRIEND_TEST(FieldTest, it_detects_points_in_field_properly);
+    FRIEND_TEST(FieldTest, it_calculates_obstacles);
+    FRIEND_TEST(FieldTest, penalty_points);
+    FRIEND_TEST(FieldTest, goal_angle);
+
+private:
+    // Used to convert protobuf names to field names.
+    std::map<std::string, std::string> NAME_MAP = {
```

Comment:
Make this const, as it's defined on compilation time.

Reply:
Doing so will result in an error, so I did not change it to const.